internal/acs/runner.go

```diff
@@ -0,0 +1,250 @@
+					}
+
+					index := ConvertLabelsToString(labels)
+					level.Info(logger).Log("msg", "Ensuring metric", "index", index, "cvss", val.cvss)
+					metric := prometheus.MustNewConstMetric(
+						collector.Metric,
+						prometheus.GaugeValue,
+						val.cvss,
+						cveSummaryData.clusterName,
+						key,
+						val.severity,
+						strconv.FormatBool(val.isFixable),
+						strconv.FormatInt(int64(len(cveSummaryData.instances)), 10),
+						cveSummaryData.namespaceName,
+						cveSummaryData.imageRemote,
+						cveSummaryData.imageRegistry,
+						cveSummaryData.imageTag,
+						cveSummaryData.imageID,
+						cveSummaryData.operatingSystem,
+					)
+					metricMap[index] = metric
+				}
+			}
+		}
+
+	}
+	return metricMap
+}
+
+func processClusterCVEs(cluster acsclient.Cluster, client acsclient.Client) (map[string]cveSummaryData, error) {
+	query := acsclient.Query{
+		"Cluster": cluster.Name,
+	}
+	exportDeployments, err := client.GetExportDeployments(query)
+	if err != nil {
+		return nil, fmt.Errorf("errors getting export deployments:: %v", err)
+	}
+	exportImages, err := client.GetExportImages(query)
+	if err != nil {
+		return nil, fmt.Errorf("errors getting export images:: %v", err)
+	}
+	exportDeploymentsMap := getexportDeploymentsMap(exportDeployments)
+	exportImagesMap := getExportImagesMap(exportImages)
+	return processNamespaces(query, client, exportDeploymentsMap, exportImagesMap)
+}
+
+func getExportImagesMap(exportImages []acsclient.ExportImage) map[string]acsclient.ExportImage {
+	imagesMap := make(map[string]acsclient.ExportImage)
+	for _, exportImage := range exportImages {
+		imagesMap[exportImage.ID] = exportImage
+	}
+	return imagesMap
+}
+
+func getexportDeploymentsMap(exportDeployments []acsclient.ExportDeployment) map[string][]acsclient.ExportDeployment {
+	exportDeploymentsMap := make(map[string][]acsclient.ExportDeployment)
+	for _, exportDeployment := range exportDeployments {
+		deployments, exists := exportDeploymentsMap[exportDeployment.Namespace]
+		if exists {
+			deployments = append(deployments, exportDeployment)
+			exportDeploymentsMap[exportDeployment.Namespace] = deployments
+		} else {
+			exportDeploymentsMap[exportDeployment.Namespace] = []acsclient.ExportDeployment{exportDeployment}
+		}
+	}
+	return exportDeploymentsMap
+}
+func processNamespaces(query acsclient.Query, client acsclient.Client, exportDeploymentsMap map[string][]acsclient.ExportDeployment, exportImagesMap map[string]acsclient.ExportImage) (map[string]cveSummaryData, error) {
+	namespaces, err := client.GetNamespaces(query)
+	if err != nil {
+		return nil, fmt.Errorf("errors getting namespaces:: %v", err)
+	}
+	cveMap := map[string]cveSummaryData{}
+	for _, namespace := range namespaces {
+		query["Namespace"] = namespace.Metadata.Name
+		deployments := exportDeploymentsMap[namespace.Metadata.Name]
+		if len(deployments) > 0 {
+			summarizeCVEData(query, cveMap, deployments, exportImagesMap)
+		}
```

Comment:
can we inline `summarizeCVEData` here since iterate `deployments` will skip summarize step if it's empty.

```golang
		deployments := exportDeploymentsMap[namespace.Metadata.Name]
		for _, deployment := range deployments {
			populateCveMap(query, deployment, cveMap, exportImagesMap)
		}
```